### DISCUSSION TITLE
Headings same size as text (using `em`)

### DISCUSSION THREAD

**Discussion by yochem** [typst/typst#2919] [Q&A] [answered]
Why doesn't the following work:

```typst
#set text(12pt)

#show heading: text.with(size: 1em, weight: "bold")

= Test

#lorem(10)
```

![test](https://github.com/typst/typst/assets/23235841/f3306d90-b287-4a3b-8d4a-ff54d327b10e)

In my understanding, `1em` would mean the same size as the general text size. Or is it `1 * <default heading size>`?

**Comment by laurmaedje** [comment]
1em stacks like you said. In the way, you've written it (as a function that transforms) an existing heading, this means that 1em is applied to the existing heading size.

**Reply by yochem** [reply]
So is it possible to set the heading the same size as the text, without knowing the text size? (Context: I'm writing an APA7 template, where the headings should be the same size as the text, and want users to be able to set their own text size without parameter to the template)

**Reply by laurmaedje** [reply, marked as answer]
Currently, it is not really possible to get rid of the default heading scaling (which is partly a bug). If you're fine with a hacky solution, you can set the inverse of the default heading scaling as em-size. The defaults are `1.4em` for level 1, `1.2em` for level 2, and `1em` for everything else.

**Reply by yochem** [reply]
What also works is the following:

```typst
// template.typ
#let apa7(...) = {
  let fontsize = state("fontsize", 0pt)

  style(styles => {
    fontsize.update(measure(v(1em), styles).height)
  })

  show heading: it => {
    text(fontsize.at(it.location()), weight: "bold", it)
  }
}

// main.typ
#import "template.typ": apa7
#show text(<adjust textsize here>)
#show apa7.with(...)
```
